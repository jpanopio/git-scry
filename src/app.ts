import config from 'config';
import express from 'express';
import session from 'express-session';
import bodyParser from 'body-parser';
import { APP_SESSION_SECRET } from './constants/env';
import router from './router';
import logger from './logger';

const app = express();

app.use(
  ({ originalUrl, method }, _, next) => {
    logger.info(`${method}: ${originalUrl}`);
    next();
  }
);

app.use(
  session({
    name: 'id',
    secret: APP_SESSION_SECRET,
    resave: false,
    saveUninitialized: false,
  })
);

app.set('port', config.get('port'));
app.use(bodyParser.json());
app.use('/', router);

export default app;
