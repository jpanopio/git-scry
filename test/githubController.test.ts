import request from 'supertest';
import app from '../src/app';

describe('GitHub Controller', () => {
  it('redirects the user to the GitHub login page', async () => {
    const { text: redirectLink } = await request(app).get('/github-login');

    expect(redirectLink).toMatchSnapshot();
  });
});
