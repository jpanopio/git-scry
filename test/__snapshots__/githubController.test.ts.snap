// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitHub Controller redirects the user to the GitHub login page 1`] = `"Found. Redirecting to https://github.com/login/oauth/authorize?client_id=undefined"`;
